### PR TITLE
Address visual QA for vertical tabs 

### PR DESCRIPTION
**Pull request by sangwoo108**
Changes in this PR

### Group
- [x] Group name font should be bold and 13px

### Header
- [x] Update icons in header view(toggle button and search button)
- [x] toggle button in collapsed state is not aligned center
- [x] toggle button color should be match with icons on toolbar in dark mode -> white

### Tabs
- [x] Spacing around tabs is a little bit bigger (8px → 4px)
- [x] Tab's width should be 36px / when in group, 32px
- [x] Pinned tabs border radius should be 8px, while unpinned tabs has 4px radius.
- [x] Pinned tabs' borderline looks too thin
- [x] Pinned tabs' drop shadow looks clipped
- [x] Active tab's BG color is too strong when it's dark mode
- [x] Divider color in dark mode is not accurate

### New tab button
- [x] Spacing and color of new tab button should be matched to those of tabs (icon padding, title etc.)
- [x] NewTabButton's hover effect should have padding just like tabs

Resolves https://github.com/brave/brave-browser/issues/29256
Possibly fix https://github.com/brave/brave-browser/issues/29283

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: app/vector_icons/BUILD.gn [hunk 1]
```diff
@@ -55,8 +55,8 @@ aggregate_vector_icons("brave_vector_icons") {
       "sidebar_reading_list.icon",
       "sidebar_settings.icon",
       "sidebar_trash.icon",
-      "vertical_tab_collapse_button.icon",
-      "vertical_tab_expand_button.icon",
+      "vertical_tab_strip_ntb.icon",
```
Comment (fallaciousreasoning) [inline app/vector_icons/BUILD.gn:58]: Out of curiousity, are either of these icons in Leo? You could swap to using the Leo icons, and that way they'll get updated when the designers change them :)

https://leo.bravesoftware.com/?path=/story/icon--all-icons

(you can add them to the `leo_icons` array in `brave/app/vector_icons/BUILD.gn` if Leo has the ones you want)

File: browser/ui/views/frame/vertical_tab_strip_region_view.cc [hunk 2]
```diff
@@ -140,6 +148,224 @@ class CustomScrollView : public views::ScrollView {
...
+  void OnPaintFill(gfx::Canvas* canvas) const override {
+    if (tab_strip()
+            ->GetCustomBackgroundId(BrowserFrameActiveState::kUseCurrent)
+            .has_value()) {
+      BraveNewTabButton::OnPaintFill(canvas);
+      return;
+    }
+
+    auto* cp = GetColorProvider();
+    DCHECK(cp);
```
Comment (fallaciousreasoning) [inline browser/ui/views/frame/vertical_tab_strip_region_view.cc:283]: I saw a thread that Chromium's style guide has changed to prefer `CHECK` but I don't know what Brave's stance is 
https://bravesoftware.slack.com/archives/C7VLGSR55/p1679324978507629

File: browser/ui/views/tabs/brave_new_tab_button.cc [hunk 3]
```diff
@@ -74,144 +59,40 @@ SkPath BraveNewTabButton::GetBorderPath(const gfx::Point& origin,
...
-  gfx::Rect text_bounds = GetLocalBounds();
-  text_bounds.Inset(
-      gfx::Insets().set_left(kTextLeftMargin).set_right(kInsetForVerticalTab));
-  text_->SetBoundsRect(text_bounds);
-  shortcut_text_->SetBoundsRect(text_bounds);
+gfx::Insets BraveNewTabButton::GetInsets() const {
+  // Give an additional left margin to make more space from tab.
+  // TabStripRegionView::UpdateNewTabButtonBorder() gives this button's inset.
+  // So, adding more insets here is easy solution.
+  return NewTabButton::GetInsets() + gfx::Insets::TLBR(0, 6, 0, 0);
```
Comment (fallaciousreasoning) [inline browser/ui/views/tabs/brave_new_tab_button.cc:97]: Is this the same `6` that is used [here](https://github.com/brave/brave-core/pull/17732/files#diff-5b35e9275153a15ac587b50c869187b3def209db671eb7c8e6cbbc2b6512fcafR263)? Maybe it's worth putting them in a variable?

File: browser/ui/views/tabs/tab_drag_controller.cc [hunk 4]
```diff
@@ -83,24 +83,19 @@ gfx::Point TabDragController::GetAttachedDragPoint(
 
 void TabDragController::MoveAttached(const gfx::Point& point_in_screen,
                                      bool just_attached) {
-  TabDragControllerChromium::MoveAttached(point_in_screen, just_attached);
   if (!is_showing_vertical_tabs_) {
+    TabDragControllerChromium::MoveAttached(point_in_screen, just_attached);
```
Comment (fallaciousreasoning) [inline browser/ui/views/tabs/tab_drag_controller.cc:87]: I think this call is the same in both paths? Maybe it can be moved back out?

**Review by fallaciousreasoning — APPROVED** [review]
LGTM @sangwoo108 - sorry for the slow review, I was off last week

Just checking the New Tab button looks right on Linux (this is hovering):
image

Unrelated to this PR (I think) but opening Tab Search via the shortcut while vertical tabs are collapsed opens the bubble in the wrong place:
image

I think there's a bug where the bottom shadow sometimes isn't removed:
image
sorry, I don't remember how I got into this state, and I'm struggling to repro it

**Review by simonhong — APPROVED** [review]

File: browser/ui/views/tabs/brave_new_tab_button.cc (hunk 3 above)
Comment (sangwoo108) [reply browser/ui/views/tabs/brave_new_tab_button.cc:97]: Sorry, the link doesn't seem to lead me to the right place. Could you give me another pointer?

File: app/vector_icons/BUILD.gn (hunk 1 above)
Comment (sangwoo108) [reply app/vector_icons/BUILD.gn]: Great! Done

File: browser/ui/views/tabs/tab_drag_controller.cc (hunk 4 above)
Comment (sangwoo108) [reply browser/ui/views/tabs/tab_drag_controller.cc]: Oh yeah, you're right.

File: browser/ui/views/frame/vertical_tab_strip_region_view.cc (hunk 2 above)
Comment (sangwoo108) [reply browser/ui/views/frame/vertical_tab_strip_region_view.cc]: Oh, missed that. Thanks for letting me know that!

**Comment by sangwoo108** [comment]
> Just checking the New Tab button looks right on Linux (this is hovering):

Oops, I made a mistake. Fixed.

File: browser/ui/views/tabs/brave_new_tab_button.cc (hunk 3 above)
Comment (fallaciousreasoning) [reply browser/ui/views/tabs/brave_new_tab_button.cc:97]: Hmm, that's weird - I think the Github link is broken. When I tried getting it again it gave me the same link and it definitely doesn't take you to the right place :laughing: 

Does this one work?
https://github.dev/brave/brave-core/blob/faf5ed87c5429c3a7ab7fc9c453f637cf2d3ea42/browser/ui/views/frame/vertical_tab_strip_region_view.cc#L263

**Review by fallaciousreasoning — APPROVED** [review]
Still LGTM :smile:

File: browser/ui/views/tabs/brave_new_tab_button.cc (hunk 3 above)
Comment (sangwoo108) [reply browser/ui/views/tabs/brave_new_tab_button.cc:97]: Awesome! Unfortunately, that inset is a coincidence.

**Comment by sangwoo108** [comment]
> I think there's a bug where the bottom shadow sometimes isn't removed:

I've yet to reproduce this, I think but I've seen this myself. I'll keep my eyes on it! Thanks

**Pull request merged**